### PR TITLE
mbed OS quickstart 

### PR DESCRIPTION
**Pull request by BlackstoneEngineering**
This is the initial draft of the mbed os quickstart. I have chosen to over write the first application tutorial as this content is made redundant by the quickstart. 

Please note that this does not include the docs.json changes as I was advised by @AnotherButler  to not make those changes but request them to be made on my behalf. I would like to request the section be renamed from 'Your First Application' to `Quickstart`.

### CODE REVIEW

**Comment by BlackstoneEngineering** [comment]
@AnotherButler , do not merge, do not review. Still pending input from os tech and product leads, content will likely change slightly.

@sg- any comments please make as changes here

@senthilr  for tracking

**Comment by sg-** [comment]
Can I see this rendered anywhere?

**Comment by BlackstoneEngineering** [comment]
@sg- [mbed-os-qs-rough-draft.zip](https://github.com/ARMmbed/Handbook/files/1568686/mbed-os-qs-rough-draft.zip)
Unfortunantly there is no public preview / notebook type mechanism in the new docs system, as a result i had to build them locally. In the above zip file is the rendered pages.

**Comment by iriark01** [comment]
There actually is a preview mechanism. You just need to ask.

**Comment by BlackstoneEngineering** [comment]
@iriark01 need a mechanism for preview that does not require external interraction and that cannot be overwritten by any other than the user. Something like the notebook pages fromt he old system would be great.

**Comment by AnotherButler** [comment]
@sg- You can view it here: https://docs.internal.mbed.com/docs/master/quickstart/index.html

**Comment by BlackstoneEngineering** [comment]
@AnotherButler , added changes / feedback from Sam G, please proceed with review

**Comment by AnotherButler** [comment]
Nice work, @yennster I'm looking over this now, but I probably won't have edits ready for you to review until the new year.

File: docs/tutorials/first_app/blinky_cli.md [hunk 1]
```diff
@@ -1,128 +1,165 @@
...
-Allocated Stack: unknown
-Total Static RAM memory (data + bss): 7168 bytes
-Total RAM memory (data + bss + heap + stack): 7168 bytes
-Total Flash memory (text + data + misc): 43402 bytes
-Image: .\.build\K64F\ARM\mbed-os-example-blinky.bin             
-```
+**Note** : you can get the name of the board plugged into your computer by running `mbed detect` and you can get a full list of supported toolchains / targets by running the `mbed compile --supported`
+
+
+### Debug
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_cli.md]: Query: Should debugging be part of the quickstart? It feels out of scope to me.

File: docs/tutorials/first_app/blinky_cli.md [hunk 2]
```diff
@@ -1,128 +1,165 @@
...
-The program file, `mbed-os-example-blinky.bin`, is under your `mbed-os-example-blinky\build\K64F\ARM\` folder.
+Invoke `mbed detect` from your command line to determine which communication port your board is connected to (i.e. `COM18`, `/dev/ttyACM0`, etc.). Unless otherwise specified, `printf` defaults to a baud rate/speed of `9600` on mbed OS.
 
-### Programming your board
+### Further Reading
 
-Arm Mbed Enabled boards are programmable by drag and drop over a USB connection:
+- Documentation
+  - [Mbed OS API's](https://os.mbed.com/docs/v5.6/reference/apis.html) - List of all API's available in Mbed OS
+  - [Peripheral Drivers](https://os.mbed.com/docs/v5.6/reference/drivers.html) - IO Driver API's (I2C, SPI, UART, ... etc)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_cli.md]: Query: Why do we link to the APIs and then link to a subsection of the APIs? They're included in the link above.

File: docs/tutorials/first_app/blinky_compiler.md [hunk 3]
```diff
@@ -1,70 +1,81 @@
...
+Open the folder where the executable file was downloaded, then click and drag (or copy and paste) the file to your mbed board's USB device folder.
 
-The imported Blinky has two interesting parts:
+Once the file has been flashed to the board, press the board's "reset" button and you should now see the LED blinking.
 
-- `main.cpp`, where the Blinky-specific code is. You can double-click the file in the navigation pane on the left to view the code.
-- `mbed-os`, where the Arm Mbed OS codebase is.
 
-Later we'll compile the code; this will take both of these parts and create a single application file from them.
+### Debug
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: Again, is debugging part of the scope of the quickstart?

File: docs/tutorials/first_app/blinky_compiler.md [hunk 4]
```diff
@@ -1,70 +1,81 @@
...
+#### Printf
 
-To add a board to your list, go to <a href="https://os.mbed.com/platforms/" target="_blank">the board's page on `os.mbed.com`</a>, and click the **`Add to your Mbed Compiler`** button:
+Another way to do basic debugging is to use the `printf` command in your code, then read the output using a serial terminal such as [PuTTY](http://www.putty.org/) or [CoolTerm](http://freeware.the-meiers.org/). For example, add `printf("Hello World!\n\r");` to the top of your main function, then recompile the program and flash it to your device.
 
-<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/add_board.png)<span>Adding a board to the Mbed Online Compiler's board list</span></span>
+Unless otherwise specified, `printf` defaults to a baud rate/speed of `9600` on mbed OS. To determine which communication port your board is connected to, follow the instructions for your operating system below:
 
-#### Selecting a board
+##### Windows
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: This section feels out of place. In the offline section, we listed environment first. Here, we list it last. Also, I don't think we should talk about using the command-line in the section about the Mbed Online Compiler.

File: docs/tutorials/first_app/blinky_compiler.md [hunk 5]
```diff
@@ -1,70 +1,81 @@
...
 
-1. Click **Compile**.
 
-	<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/compileandinstall.png)<span>The Compile menu; choose Compile to build and download your application</span></span>
+### Further Reading
 
-1. The program compiles:
+- Documentation
+	- [Mbed OS API's](https://os.mbed.com/docs/v5.6/reference/apis.html) - List of all API's available in Mbed OS
+	- [Peripheral Drivers](https://os.mbed.com/docs/v5.6/reference/drivers.html) - IO Driver API's (I2C, SPI, UART, ... etc)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: Again, why are we linking to the APIs and then linking to a subset of the APIs? They were included in the link above.

File: docs/tutorials/first_app/blinky_cli.md [hunk 6]
```diff
@@ -1,128 +1,165 @@
...
 
-This tutorial builds Blinky using the Arm Mbed CLI, which allows you to build Mbed OS applications on your own machine. You will need to install Mbed CLI and a toolchain before you can work with Blinky.
+### Setup
+Please skip to the sub section for your host OS. The setup instructions will walk you through how to get mbed CLI installed locally on your system. 
 
-<span class="tips">Please create a <a href="https://os.mbed.com/account/signup/" target="_blank">developer account</a>. It's free, and we don't spam.</span>
+[**Windows**](#windows) | [**OSX**](#mac-osx) | [**Linux**](#linux)
 
-### Blinky's code
+#### Windows
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: the .exe is the pre-reqs all wrapped up into one easy step, eventually want to have a similar thing for OSX too. but for now it doesnt exist

File: docs/tutorials/first_app/blinky_cli.md [hunk 7]
```diff
@@ -1,128 +1,165 @@
...
 
-Blinky's code is a simple `while` loop inside the `main()` function:
+##### 1. Install mbed CLI
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: its all bundled up in the .exe

File: docs/tutorials/first_app/blinky_cli.md (hunk 1 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: This is a light touch, there is a more in depth tutorial for debugging. I believe it is fine.

File: docs/tutorials/first_app/blinky_cli.md (hunk 2 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: This is based on interactions with users. majority of users still are interested in the 2.0 API's, now called peripheral drivers, so we link directly to them. Never hurts to have multiple paths

File: docs/tutorials/first_app/blinky_compiler.md (hunk 3 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: yes, this is very limited in its scope, thus i feel it is appropriate

File: docs/tutorials/first_app/first_program.md [hunk 8]
```diff
@@ -1,14 +1,12 @@
-## Your first Arm Mbed application
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/first_program.md]: @sg-

File: docs/tutorials/first_app/first_program.md [hunk 9]
```diff
@@ -1,14 +1,12 @@
...
+## Choose your environment
+In the mbed ecosystem you have choice on how you want to develop, **Online** and **Offline**. 
 
-We have an example application called Blinky that you can use to get to know Arm Mbed OS and the development tools. It's one of the simplest examples of Mbed OS. It shows the use of a DigitalOut object to represent an LED and the `wait()` call. This is good practice because if there were other threads, they could be scheduled and run while the first thread is waiting.
+1) Online there is the Mbed Compiler. This choice is great if you want an environment that 'just works'. All you need is a web browser and you're good to go.
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/first_program.md]: thats always been our pitch. @sg- to see if thats changed

File: docs/tutorials/first_app/first_program.md [hunk 10]
```diff
@@ -1,14 +1,12 @@
-## Your first Arm Mbed application
+## Choose your environment
+In the mbed ecosystem you have choice on how you want to develop, **Online** and **Offline**. 
 
-We have an example application called Blinky that you can use to get to know Arm Mbed OS and the development tools. It's one of the simplest examples of Mbed OS. It shows the use of a DigitalOut object to represent an LED and the `wait()` call. This is good practice because if there were other threads, they could be scheduled and run while the first thread is waiting.
+1) Online there is the Mbed Compiler. This choice is great if you want an environment that 'just works'. All you need is a web browser and you're good to go. 
 
-You can use either of these tools:
+2) Offline there is Mbed CLI, a full suite of tools for use on the command line. Mbed CLI is compatible with Windows, Linux and OSX. This option provides more configuratino options but also requires slightly more setup.
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/first_program.md]: Yes. mbed CLI is the command line interface for all things mbed.

File: docs/tutorials/first_app/blinky_compiler.md (hunk 5 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: see previous answer

File: docs/tutorials/first_app/blinky_compiler.md (hunk 4 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: Im not sure I understand what you are trying to say, lets discuss in person

File: docs/tutorials/first_app/blinky_cli.md [hunk 11]
```diff
@@ -1,128 +1,165 @@
-## Blinky on Arm Mbed CLI
+## Offline - Mbed CLI
 
-This tutorial builds Blinky using the Arm Mbed CLI, which allows you to build Mbed OS applications on your own machine. You will need to install Mbed CLI and a toolchain before you can work with Blinky.
+### Setup
```
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: lets discuss in person

[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: Reviewed togetehr, amanda to re-write sentence and git 'er done.

File: docs/tutorials/first_app/blinky_cli.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: Reviewed, amanda to add sentence and push. Git'er'done

File: docs/tutorials/first_app/blinky_cli.md (hunk 2 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_cli.md]: agree to disagree and let it slide.

File: docs/tutorials/first_app/first_program.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/first_program.md]: change to 'mbed CLI is the tool for using Mbed on thec ommand line"

File: docs/tutorials/first_app/blinky_compiler.md [hunk 12]
```diff
@@ -1,70 +1,81 @@
...
-To get Blinky into the Mbed Online Compiler, click the **`Import into Mbed IDE`** button below:
+#### Setup environment
 
-[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-blinky/)](https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-blinky/file/tip/main.cpp)
+- Plug your Mbed board into your computer, and open its USB device folder.
+- Double click on the `MBED.HTM` file. (This adds your Mbed platform to the Online Compiler.)
 
-You're taken to the online IDE, and the **Import Program** dialog box opens:
+If you do not have an Mbed board, go to [os.mbed.com/platforms](http://os.mbed.com/platforms), select a board and click the “Add to your Mbed Compiler” button.
+<span class="images">![](https://sarahmarshy.github.io/img/add-to-compiler.png)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: This image is out of date. Please update it and upload it to the "images" folder at a higher level of the Handbook.

File: docs/tutorials/first_app/blinky_compiler.md [hunk 13]
```diff
@@ -1,70 +1,81 @@
...
-<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/import_dialog.png)<span>Importing Blinky</span></span>
+### Code
 
-The import mechanism offers a default name, but you're free to change it. When you're done, click **Import**.
+#### Import 
 
-### Viewing Blinky
+Visit the Mbed OS [blinky example repository](https://developer.mbed.org/teams/mbed-os-examples/code/mbed-os-example-blinky/), and click the "Import into Compiler" button.
+ 
+<span class="images">![](https://sarahmarshy.github.io/img/import-compiler.png)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: This image is out of date. Please update it and upload it to the "images" folder at a higher level of the Handbook.

File: docs/tutorials/first_app/blinky_compiler.md [hunk 14]
```diff
@@ -1,70 +1,81 @@
...
 
-The imported Blinky has two interesting parts:
+#### Compile
 
-- `main.cpp`, where the Blinky-specific code is. You can double-click the file in the navigation pane on the left to view the code.
-- `mbed-os`, where the Arm Mbed OS codebase is.
+Click on the "Compile" button. Your browser downloads the program as an executable file.
 
-Later we'll compile the code; this will take both of these parts and create a single application file from them.
+<span class="images">![](https://sarahmarshy.github.io/img/compile.png)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: This image is out of date. Please update it and upload it to the "images" folder at a higher level of the Handbook.

File: docs/tutorials/first_app/blinky_compiler.md [hunk 15]
```diff
@@ -1,70 +1,81 @@
...
+### Debug
 
-To add a board to your list, go to <a href="https://os.mbed.com/platforms/" target="_blank">the board's page on `os.mbed.com`</a>, and click the **`Add to your Mbed Compiler`** button:
+#### Desktop IDE
 
-<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/add_board.png)<span>Adding a board to the Mbed Online Compiler's board list</span></span>
+To debug using a desktop IDE such as Keil uVision, IAR or Eclipse, click the `Export` button under `Program Details`. Select your export platform and IDE, and click `Export`. Your browser downloads a `.zip` file with the project files.
 
-#### Selecting a board
+<span class="images">![](https://sarahmarshy.github.io/img/export.png)
```
Comment (AnotherButler) [inline docs/tutorials/first_app/blinky_compiler.md]: This image is out of date. Please update it and upload it to the "images" folder at a higher level of the Handbook.

**Comment by AnotherButler** [comment]
Thanks. Please be sure to upload updated images as requested. Other than that, I believe my remaining question and request (confirming that all one needs for CLI is a browser and asking for an introductory paragraph) are the ones we agreed I need to speak to product and positioning experts about.

File: docs/tutorials/first_app/blinky_compiler.md (hunk 12 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: @yennster can you please update the image?
@AnotherButler where would you like us to upload the image? should we create a `img` folder or is there a master image folder we should upload to?

File: docs/tutorials/first_app/blinky_compiler.md (hunk 13 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: @yennster can you please update the image?
@AnotherButler where would you like us to upload the image? should we create a `img` folder or is there a master image folder we should upload to?

File: docs/tutorials/first_app/blinky_compiler.md (hunk 14 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: @yennster can you please update the image?
@AnotherButler where would you like us to upload the image? should we create a `img` folder or is there a master image folder we should upload to?

File: docs/tutorials/first_app/blinky_compiler.md (hunk 15 above)
Comment (BlackstoneEngineering) [reply docs/tutorials/first_app/blinky_compiler.md]: @yennster can you please update the image?
@AnotherButler where would you like us to upload the image? should we create a `img` folder or is there a master image folder we should upload to?

File: docs/tutorials/first_app/blinky_compiler.md (hunk 12 above)
Comment (AnotherButler) [reply docs/tutorials/first_app/blinky_compiler.md]: As mentioned previously, there's an "images" folder at a high level in our documentation where we put all images: https://github.com/ARMmbed/Handbook/tree/new_engine/docs/images

File: docs/tutorials/first_app/blinky_compiler.md (hunk 13 above)
Comment (AnotherButler) [reply docs/tutorials/first_app/blinky_compiler.md]: As mentioned previously, there's an "images" folder at a high level in our documentation where we put all images: https://github.com/ARMmbed/Handbook/tree/new_engine/docs/images

File: docs/tutorials/first_app/blinky_compiler.md (hunk 14 above)
Comment (AnotherButler) [reply docs/tutorials/first_app/blinky_compiler.md]: As mentioned previously, there's an "images" folder at a high level in our documentation where we put all images: https://github.com/ARMmbed/Handbook/tree/new_engine/docs/images

File: docs/tutorials/first_app/blinky_compiler.md (hunk 15 above)
Comment (AnotherButler) [reply docs/tutorials/first_app/blinky_compiler.md]: As mentioned previously, there's an "images" folder at a high level in our documentation where we put all images: https://github.com/ARMmbed/Handbook/tree/new_engine/docs/images

File: docs/tutorials/first_app/first_program.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/tutorials/first_app/first_program.md]: @yennster Could you please help me create an intro paragraph? Or sit with me and verbalize what the user should expect to get out of this? Thanks so much.

**Comment by iriark01** [comment]
I don't think the quick start should have Mbed CLI (which, last we checked, can take up to two hours to install on a Mac). Don't confuse people with two options - take them straight to the IDE, which is the quickest way to get started, and leave the Mbed CLI bit for later. Make it explicitly about Mbed CLI rather than the Mbed OS quick start.

**Comment by senthilr** [comment]
Thinking about this a bit, I think it makes more sense to present both Online ide and CLI options upfront. It is clearly stated that Online ide is the easiest option that "just works" whereas offline provides more flexibility but does require some effort to set up. 
@sg-

**Comment by sg-** [comment]
Is there a rendered version of this work? It's really hard to review the flow from the change set.

**Comment by iriark01** [comment]
@senthilr I remind you of the context - this isn't going to be part of the handbook, it's going to be a one-page quick start that's way higher in the site's hierarchy than the handbook. It is not the place to get into multiple processes, especially ones that take more than five minutes.

**Comment by AnotherButler** [comment]
@sg- You can view the rendered documents here: https://docs.internal.mbed.com/docs/master/quickstart/index.html

File: docs/tutorials/first_app/blinky_compiler.md (hunk 15 above)
Comment (yennster) [reply docs/tutorials/first_app/blinky_compiler.md]: @AnotherButler @BlackstoneEngineering Sorry, I just saw this note today. Do you still need those images uploaded?

Comment (AnotherButler) [reply docs/tutorials/first_app/blinky_compiler.md]: @yennster Yes, please. If you could upload these at your convenience, that would be great. Thanks for checking.

Comment (yennster) [reply docs/tutorials/first_app/blinky_compiler.md]: @AnotherButler will do later today!

**Comment by yennster** [comment]
@AnotherButler images are uploaded 👍

File: docs/tutorials/first_app/first_program.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (yennster) [reply docs/tutorials/first_app/first_program.md]: @AnotherButler Do you still need an intro paragraph before `## Choose your environment`?

[the remark this answers is not in the document]
Comment (AnotherButler) [reply docs/tutorials/first_app/first_program.md]: @yennster Yes. I was waiting to write it until we knew whether we'd be including Mbed CLI, but it may not be a bad idea to start drafting something that we could use whether or not we include CLI.

**Comment by BlackstoneEngineering** [comment]
Can we please get this published?

**Comment by iriark01** [comment]
I will catch up with Simon F this week

**Comment by BlackstoneEngineering** [comment]
@iriark01 why does Simon F come into this? Sam G is the content owner and we have his approval to proceed. I would really like to get this published ASAP as it is going on 2 months delayed.

**Comment by iriark01** [comment]
Amanda, you can merge this for now if you want. Simon F and I will continue off line.

**Comment by yennster** [comment]
@AnotherButler I fixed those merge conflicts!

**Comment by AnotherButler** [comment]
@yennster Go you. Thanks so much. Merging now.
